python/src/spark_rapids_ml/umap.py

```diff
@@ -596,9 +739,76 @@ def _out_schema(self, input_schema: StructType) -> Union[StructType, str]:
         )
 
     def get_model_attributes(self) -> Optional[Dict[str, Any]]:
-        """Override parent method to bring broadcast variables to driver before JSON serialization."""
-        if not isinstance(self.embedding_, list):
+        """
+        Override parent method to bring broadcast variables to driver before JSON serialization.
+        """
+        if not isinstance(self.embedding_, np.ndarray):
             self._model_attributes["embedding_"] = self.embedding_.value
-        if not isinstance(self.raw_data_, list):
+        if not isinstance(self.raw_data_, np.ndarray):
```

Comment:
These could be moved to an internal accessor API (if modifying API per above).

Reply:
I think this is needed in case the user never accesses the attributes before saving - attributes will need to be retrieved and stored from the broadcast.